### ISSUE TITLE
I'm trying to implement a third map

### ISSUE THREAD

**Issue by HolySSSS** [TimPushkin/DepNav#81]
Hi. This issue might be like a guide for other onlookers and especially programmer newbies, if you are wondering about using this project for your own purposes. Refer to the /tools inside the app, use WSL or VM Linux to execute the .sh file and tinker with the .py files, you might have to change the script to just manually write down the file instead of parsing it. You do have to make a JSON file yourself, but it's pretty easy in principle, you just write down the X and Y value of any point by checking them with any graphical software in the map image into a new object in JSON.

So far I'm stuck after this point. I might solve it by myself, maybe not, we will see. Now let's get to the problem.

I'm trying to implement a third map for the purposes getting used to understanding someone's code by myself, but this so far stumped me.

I added a new map into the same Database, a new tile pyramid into the app and it leads to the app crashing in Android Studio.
So, there are 3 maps, 3 tile sets now in the DepNav, nothing else.
I did not do any other actions to the Database or the tile sets after creating them and putting them into the app.

![изображение](https://github.com/TimPushkin/DepNav/assets/81934438/e79b2a4a-f65d-432b-b267-4993c7af8646)
![изображение](https://github.com/TimPushkin/DepNav/assets/81934438/b0ad666f-c115-497b-9a2b-e948bed52aae)

I also did not add any new UI elements to represent the third map at the moment of writing this, for I am currently trying to understand how to implement one in your code.
The third map is called "Third" or "Третий" currently, as in internalName (for others, check JSONs to understand what's that) and the actual name of the map.
Thanks for any help.

_Originally posted by @HolySSSS in https://github.com/TimPushkin/DepNav/issues/79#issuecomment-2113618544_

**Comment by HolySSSS** [comment]
I managed to make the error much smaller by adding a copy of a XML file and a XML.FLAT file, so far I'm having trouble linking them.

![изображение](https://github.com/TimPushkin/DepNav/assets/81934438/a4bf34d8-5eea-4660-b3da-797f47dc798d)

**Comment by HolySSSS** [comment]
I managed to add the UI element by creating a copy of "logo_spbu_mm" and renaming it "logo_third" and then I pasted it into folder "drawable" in the folder "res" inside the Android Studio. I pasted into "drawable", not "drawable-v24".

The problem wasn't fixed and seems to stay the same.

![изображение](https://github.com/TimPushkin/DepNav/assets/81934438/6b1a8af9-0ea7-40ea-b876-f465891b7e44)

**Comment by HolySSSS** [comment]
Currently assuming two options:
a) the app can only recognise what is already in.
b) the app can't support 3 or more maps in a DB.

**Comment by HolySSSS** [comment]
Addendum:
The map "Third" features 7 floors.

That's all info I can give for now.

**Comment by TimPushkin** [comment]
Hey, thank you for your interest in DepNav. I'll write instructions describing the whole process of adding a new map later today.

**Comment by HolySSSS** [comment]
As far as I can tell, I missed adding a json [ru.spbu.depnav.data.db.AppDatabase/9.json](https://github.com/TimPushkin/DepNav/pull/79/files#diff-1618dedb26f28e685fead4c4d9ff50a42f299cf2c5da96dae1d3a63e79327f1b) from this pull request #79 that was added along with a new db and json.

Gonna check the problem right now.

**Comment by HolySSSS** [comment]
Correct, I managed to fix the problem, it is now working as intended.

**Comment by HolySSSS** [comment]
You may close the issue now if you see it as fit, but here's the general gist of what you need to do:

a) Make a map in a format of WebP. Use Paint.net for example.
b) Start making a JSON file by taking the X and Y values of all points for icons. Check the JSON files in the app.
c) Use the create_tiles.sh by using a WSL or a VM Linux, whatever you prefer, by putting the map images and the script in 1 folder.
d) Use the create_map_db.py to convert the JSON file into a sqlite3 database.
e) Integrate the new DB, the new JSON and the new tile pyramid of the map into the app.

It is up to the programmer to understand what goes where in the folders.

**Comment by TimPushkin** [comment]
I've added a [wiki page](https://github.com/TimPushkin/DepNav/wiki/Creating-a-map) describing the process of map creation and also somewhat improved the docs of development scripts.

The instruction kinda resembles what you've already found out by yourself, though I don't really understand how adding `spbu-pf.json` somewhere might have helped with your problem. The app does not use JSONs directly, they are just an intermediate step of creating the database because they are easier to write by hand and track through version control.

The problem in your original post was caused by the missing map's logo so it should have been fixed right after you added the logo.
